### PR TITLE
Allow marking of new releases as prerelease

### PR DESCRIPTION
**Pull request by mathias-baumann-sociomantic**
Refs sociomantic-tsunami/neptune#148

### CODE REVIEW

File: src/octod/api/releases.d [hunk 1]
```diff
@@ -24,11 +24,12 @@ import octod.core;
         tag  = tag used for the release
         title = title for the release
         content = content of the release
+        prerelease = true if this is a prerelease, else false
 
 *******************************************************************************/
 
 public void createRelease ( ref HTTPConnection connection, ref Repository repo,
-                            string tag, string title, string content )
+    string tag, string title, string content, bool prerelease = false )
```
Comment (mihails-strasuns-sociomantic) [inline src/octod/api/releases.d]: https://dlang.org/library/std/typecons/flag.html makes such API usage much more readable :P

Comment (mathias-baumann-sociomantic) [reply src/octod/api/releases.d]: Fair point

Comment (mathias-baumann-sociomantic) [reply src/octod/api/releases.d]: While I approve of this, the whole feature feels like it's hacking around around the lack of language support for just that

**Comment by mathias-baumann-sociomantic** [comment]
Updated with Flags

File: src/octod/api/releases.d [hunk 2]
```diff
@@ -41,6 +45,7 @@ public void createRelease ( ref HTTPConnection connection, ref Repository repo,
     json["name"] = title;
     json["body"] = content;
     json["target_committish"] = tag;
+    json["prerelease"] = prerelease == Yes.prerelease ? true : false;
```
Comment (mihails-strasuns-sociomantic) [inline src/octod/api/releases.d]: `json["prerelease"] = (prerelease == Yes.prerelease);` ?

Comment (mathias-baumann-sociomantic) [reply src/octod/api/releases.d]: .. and your question is what?

Comment (mathias-baumann-sociomantic) [reply src/octod/api/releases.d]: Oh.. yeah, I see. I suppose my version is more verbose and thus easier to understand at a first glance?

**Comment by mathias-baumann-sociomantic** [comment]
Updated to simplify assignment

File: src/octod/api/releases.d (hunk 2 above)
Comment (mihails-strasuns-sociomantic) [reply src/octod/api/releases.d]: Maybe, though for me it is causing unnecessary "wait, what is going on here?" moment :)

**Review by mihails-strasuns-sociomantic — APPROVED** [review]

**Pull request merged**